Feature/update gapi client

s0ph0s-dog opened the pull request:
Resolves #73, as long as someone tests it with JBot's API key first to make sure I didn't accidentally break something I can't test…

s0ph0s-dog committed:
Update Google API Client to the latest version and attempt to fix the resulting bugs

s0ph0s-dog committed:
Update Google API Client to the latest version and attempt to fix the resulting bugs

s0ph0s-dog committed:
Add a NullCalendar for when the Google API client is misconfigured

s0ph0s-dog committed:
Add a NullCalendar for when the Google API client is misconfigured

s0ph0s-dog committed:
Add Cinch's rate limiting configuration options

rikai committed:
Merge pull request #83 from s0ph0s-2/BUGFIX/ratelimit

Add Cinch's rate limiting configuration options

rikai commented:
Going to wait to merge this until i can find a chance to test it

s0ph0s-2 commented:
I should also note that I was only able to upgrade to 0.10.x because 0.11.x depends on `mime-types` version 3-something, but `mail` depends on version 1.16. That might become an issue at some point in the future.

rikai commented:
Mail is used in the backup scripts, which is actually code that has never been touched, and currently not actively used, at least by me. It's defined in the Gemfile & you can look at the gemfile.lock to see what is requiring something. 👍 

TL;DR if it's not defined in the gemfile explicitly, it's pulled in as a dependency. If you're wanting to/up for doing an update of the gemfile dependencies (and perhaps removing some of the hard versions defined as a bonus), i would certainly not object, it's just a bit time-consuming to update and figure out if anything breaks. :)

s0ph0s-2 commented:
That's something I'll definitely consider doing, but I agree that it will be time consuming and somewhat tedious. It might also be worthwhile to go through the entire project to delete code and dependencies that are unused. Less code, fewer bugs!

This seems like something that deserves it's own issue.

rikai commented:
Honestly, the back of stuff is the only code that is completely unused as far as I remember, and that's mostly just unused because I haven't gotten around to using it rather than because it is useless but I agree, definitely is a separate issue

s0ph0s-dog committed:
Add cinch-cooldown to the Gemfile. Require at least v1.2.0, since earlier versions don't work with cinchize

s0ph0s-dog committed:
Add an example cooldown configuration to cinchize.yml.example

s0ph0s-dog committed:
Enforce cooldowns in all plugins except admin.rb and suggest.rb, since those should respond ASAP

s0ph0s-2 committed:
Merge branch 'devel' into FEATURE/add-cinch-cooldown

rikai commented:
Hm, how would you feel about updating this to 0.11 regardless, making a note for later testing/a FIXME in the backup code/the commit message? Since it's code that's not actively used, i don't mind breaking it for the sake of updating code that is actively used and fixing it later when we actually have the time and need it. 👍

rikai committed:
Merge pull request #85 from s0ph0s-2/FEATURE/add-cinch-cooldown

Add's cooldown support via cinch-cooldown
Closes #40

s0ph0s-2 commented:
I'll hunt down the backup code and see how it uses `mail`. If updating doesn't actually break anything, I'll just go ahead and update all the way to 0.11. If it does, I'll disable the rake trigger and leave a FIXME comment.

rikai committed:
Minimal implementation of discord

This is just hooking the discordrb bot into the showbot startup process
and implementing a minimal bot setup

thefirstofthe300 committed:
Added shows_test.rb

thefirstofthe300 committed:
Fix errors thrown on startup due to refactoring

thefirstofthe300 committed:
Add a working travis.yml file to allow for Travis-CI integration.

This commit provides a working Travis CI configuration file; however,
builds currently fail because, well, the test suite isn't in great
shape.

thefirstofthe300 committed:
Add a working travis.yml file to allow for Travis-CI integration.

This commit provides a working Travis CI configuration file; however,
builds currently fail because, well, the test suite isn't in great
shape.

thefirstofthe300 committed:
Update README to reflect the current minimum version of Ruby and add the build status badge

thefirstofthe300 committed:
Update README to reflect the current minimum version of Ruby and add the build status badge

thefirstofthe300 committed:
There are existing badges, dumb-dumb

thefirstofthe300 committed:
There are existing badges, dumb-dumb

rikai committed:
Merge pull request #90 from thefirstofthe300/travis-ci

Add .travis.yml

thefirstofthe300 committed:
Update Gemfile.lock to satisfy Travis

rikai committed:
Merge pull request #91 from thefirstofthe300/travis-ci

Update Gemfile.lock to satisfy Travis

thefirstofthe300 committed:
Update rspec unit tests

thefirstofthe300 committed:
Add rspec tests to keep the test suite uniform

thefirstofthe300 committed:
Merge branch 'devel' into shows-test

thefirstofthe300 committed:
Update Rakefile

thefirstofthe300 committed:
Added tests for quotelist.rb as well as did some refactoring to remove business logic from the model.

thefirstofthe300 committed:
Merge branch 'shows-test' of github.com:thefirstofthe300/Showbot into shows-test

thefirstofthe300 committed:
Update travis to reflect actual passing tests

rikai committed:
Merge pull request #92 from thefirstofthe300/shows-test

More testing and actually get travis building

s0ph0s-2 commented:
I just started digging into how the backup feature is supposed to work. It actually looks pretty useful! (For reference, [here is the documentation](http://backup.github.io/backup/v3/)). Is having backups of Showbot's database useful to JB, or are you backing the Showbot droplet up in a different way?

Currently, the backup code you said that you don't use is configured to back up the MySQL database to Amazon S3, provided that it has all of the information required to access the database and upload to an S3 bucket.

I don't see any issues that would prevent me from simply removing the version restriction and updating the `backup` and `mail` gems to the current versions. That should unblock updating the Google APIs to the most recent version. However, if the backups aren't a feature that we want to keep around, we should probably just axe the code right now and eliminate a small portion of future maintenance headaches.

s0ph0s-dog committed:
Remove version restriction on 'mail' to let 'google-api-client' update all the way to v13

s0ph0s-dog committed:
Remove version restriction on 'mail' to let 'google-api-client' update all the way to v13

s0ph0s-dog committed:
Fix merge conflicts due to my inability to Git

s0ph0s-dog committed:
Run bundle update again so that the Discord gems are in the lockfile

s0ph0s-dog committed:
Update rspec test for the calendar to reflect the new Google API Client behavior

s0ph0s-dog committed:
Add test for NullCalendar

s0ph0s-dog committed:
Update README version to the Ruby version I tested on

s0ph0s-2 commented:
I rebased on `devel` since some new dependencies have been added (Discord). It's now up to `google-api-client` v13.
